src/librustc_back/target/mod.rs

```diff
@@ -89,65 +75,65 @@ macro_rules! supported_targets {
 }
 
 supported_targets! {
-    ("x86_64-unknown-linux-gnu", x86_64_unknown_linux_gnu),
-    ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
-    ("i586-unknown-linux-gnu", i586_unknown_linux_gnu),
-    ("mips-unknown-linux-gnu", mips_unknown_linux_gnu),
-    ("mipsel-unknown-linux-gnu", mipsel_unknown_linux_gnu),
-    ("powerpc-unknown-linux-gnu", powerpc_unknown_linux_gnu),
-    ("powerpc64-unknown-linux-gnu", powerpc64_unknown_linux_gnu),
-    ("powerpc64le-unknown-linux-gnu", powerpc64le_unknown_linux_gnu),
-    ("arm-unknown-linux-gnueabi", arm_unknown_linux_gnueabi),
-    ("arm-unknown-linux-gnueabihf", arm_unknown_linux_gnueabihf),
-    ("armv7-unknown-linux-gnueabihf", armv7_unknown_linux_gnueabihf),
-    ("aarch64-unknown-linux-gnu", aarch64_unknown_linux_gnu),
-    ("x86_64-unknown-linux-musl", x86_64_unknown_linux_musl),
-    ("i686-unknown-linux-musl", i686_unknown_linux_musl),
-    ("mips-unknown-linux-musl", mips_unknown_linux_musl),
-    ("mipsel-unknown-linux-musl", mipsel_unknown_linux_musl),
-
-    ("i686-linux-android", i686_linux_android),
-    ("arm-linux-androideabi", arm_linux_androideabi),
-    ("armv7-linux-androideabi", armv7_linux_androideabi),
-    ("aarch64-linux-android", aarch64_linux_android),
-
-    ("i686-unknown-freebsd", i686_unknown_freebsd),
-    ("x86_64-unknown-freebsd", x86_64_unknown_freebsd),
-
-    ("i686-unknown-dragonfly", i686_unknown_dragonfly),
-    ("x86_64-unknown-dragonfly", x86_64_unknown_dragonfly),
-
-    ("x86_64-unknown-bitrig", x86_64_unknown_bitrig),
-    ("x86_64-unknown-openbsd", x86_64_unknown_openbsd),
-    ("x86_64-unknown-netbsd", x86_64_unknown_netbsd),
-    ("x86_64-rumprun-netbsd", x86_64_rumprun_netbsd),
-
-    ("x86_64-apple-darwin", x86_64_apple_darwin),
-    ("i686-apple-darwin", i686_apple_darwin),
-
-    ("i386-apple-ios", i386_apple_ios),
-    ("x86_64-apple-ios", x86_64_apple_ios),
-    ("aarch64-apple-ios", aarch64_apple_ios),
-    ("armv7-apple-ios", armv7_apple_ios),
-    ("armv7s-apple-ios", armv7s_apple_ios),
-
-    ("x86_64-sun-solaris", x86_64_sun_solaris),
-
-    ("x86_64-pc-windows-gnu", x86_64_pc_windows_gnu),
-    ("i686-pc-windows-gnu", i686_pc_windows_gnu),
-
-    ("x86_64-pc-windows-msvc", x86_64_pc_windows_msvc),
-    ("i686-pc-windows-msvc", i686_pc_windows_msvc),
-    ("i586-pc-windows-msvc", i586_pc_windows_msvc),
-
-    ("le32-unknown-nacl", le32_unknown_nacl),
-    ("asmjs-unknown-emscripten", asmjs_unknown_emscripten)
+    ("x86_64-unknown-linux-gnu", "json/x86_64-unknown-linux-gnu.json"),
```

Comment:
Now that these are just strings and all have the same name, you may be able to change the include to:

```rust
include_str!(concat!("json/", $target, ".json"))
```

And that way each target only needs to be listed here once